source/FreeRTOS_Stream_Buffer.c

```diff
@@ -53,13 +53,11 @@
  * @return The space between uxLower and uxUpper, which equals to the distance
  *         minus 1.
  */
-size_t uxStreamBufferSpace( const StreamBuffer_t * pxBuffer,
-                            const size_t uxLower,
-                            const size_t uxUpper )
+size_t uxStreamBufferSpace( const StreamBuffer_t * const pxBuffer,
+                            size_t uxLower,
+                            size_t uxUpper )
```

Comment:
Why do you remove const attribute for uxLower and uxUpper?

Reply:
In general, we prefer the const qualifier where arguments are not modified inside a function.
TL;DR
MISRA advises (17.8) that arguments should not be modified.
Richard Barry reminds us that FreeRTOS gets build on 18+ compilers and sometimes these qualifiers can cause warnings in a few compilers.  So a blanket rule is difficult to apply.

Reply:
As @n9wxu said, that may not add much value to add const to the parameter that is called by value. But it also violates MISRA rule. I'd suggest to add it back.

Reply:
@ActoryOu for consistency's sake, should all pass by value parameters be marked as const then? I guess I'm just curious why a couple of these functions have const parameters but most others dont.

Reply:
Applogies that I take this discussion into wrong direction.

After checking, I was saying `MISRA rule 8.13 -- A pointer should point to a const-qualified type whenever possible` and that does not relate to call by value parameter. So, I'm ok to remove that.

I'd suggest to add this change log into PR description to summarize what you're doing in this PR. IMHO, it's going far comparing to what it's saying.

Thank you.